### PR TITLE
Add scraping tests

### PR DESCRIPTION
**Pull request by Natim**
Based on #30

### CODE REVIEW

**Comment by glasserc** [comment]
Needs a rebase now that I updated kinto-http.py to 8.0.1..

Nice use of `parametrize`.

What happened to `requirements.txt` ? I think we still need it.

Personally I think we ought to keep the `listen_pulse` script, even if we can't use it yet (#39) . But I'll let you and @leplatrem fight that out.

I'm not crazy about the name `data_tools`, how about `importers`?

**Comment by leplatrem** [comment]
What is the status of this PR? Is it supposed to replace #30? Why is there a commit titled `Use a deterministic ID to create the records.` then?

> Personally I think we ought to keep the listen_pulse script, even if we can't use it yet (#39) . But I'll let you and @leplatrem fight that out.

So do I :)

> I'm not crazy about the name data_tools, how about importers?

I still wonder what was wrong with `ui` and `jobs`, or at least why was it necessary to change that now and make sure we «bikeshed early, bikeshed often»

File: data_tools/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Contributors
+============
+
+* Mathieu Agopian <magopian@mozilla.com>
+* Rémy Hubscher <rhubscher@mozilla.com>
+* Mathieu Leplatre <mathieu@mozilla.com>
+* Nicolas Perriault <nperriault@mozilla.com>
```
Comment (leplatrem) [inline data_tools/CONTRIBUTORS.rst]: Ethan missing :)

File: data_tools/CHANGELOG.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+This document describes changes between each past release.
+
+0.1 (unreleased)
+----------------
+
+**Initial version**
+
+- Scraping tools from archive
+- Scraping tools for systemaddons
```
Comment (leplatrem) [inline data_tools/CHANGELOG.rst]: Technically the systemaddons thing is not a scraper. So just write something short like:
* Scrape product releases and nightlies from https://archives.mozilla.org
* Fetch system addons update information from Balrog

File: data_tools/buildhub/utils.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import re
+FILE_EXTENSIONS = "zip|tar.gz|tar.bz2|dmg|apk"
```
Comment (leplatrem) [inline data_tools/buildhub/utils.py]: missing blank line

File: data_tools/buildhub/utils.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+def build_record_id(record):
+    version = record["target"]["version"]
+    if record["target"]["channel"] == 'nightly':
+        url_parts = record["download"]["url"].split('/')
+        date_parts = url_parts[8].split('-')
+        date = '-'.join(date_parts[:6])
+        version = '{}_{}'.format(date, version)
+    return '{source[product]}_{version}_{target[platform]}_{target[locale]}'.format(
+        **record, version=version).replace('.', '-').lower()
```
Comment (leplatrem) [inline data_tools/buildhub/utils.py]: This looks over complex (see comment in previous PR)

File: data_tools/buildhub/utils.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    - firefox-55.0a1.ach.win64.zip
+    - firefox-55.0a1.bn-IN.mac.dmg
+
+    And things we'll want to ignore:
+
+    - firefox-55.0a1.en-US.linux-i686.talos.tests.zip
+    - firefox-55.0a1.en-US.mac.crashreporter-symbols.zip
+    """
+    re_nightly = re.compile(r"\w+-(\d+.+)\.([a-z]+(\-[A-Z]+)?)\.(.+)\.({})$".format(
+        FILE_EXTENSIONS))
```
Comment (leplatrem) [inline data_tools/buildhub/utils.py]: nit: this line breaking is not elegant. How about splitting the regex in two parts?

File: data_tools/tests/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            "url": "https://archive.mozilla.org/pub/mobile/releases/42.0b2/android-api-9/fr/"
+            "fennec-42.0b2.fr.android-arm.apk",
+        }
+    },
+]
+
+
+# Build record_id from record
+@pytest.mark.parametrize("record", RECORDS)
+def test_build_record_id_from_url(record):
```
Comment (leplatrem) [inline data_tools/tests/test_utils.py]: The comment and the function name don't match. Remove (useless?) comment?

File: data_tools/tests/test_utils.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+WRONG_RELEASE_FILENAMES = [
+    ("firefox", "firefox-1.5.0.5.tar.gz.asc"),
+    ("firefox", "firefox-52.0.win32.sdk.zip"),
+]
+
+
+@pytest.mark.parametrize("product,filename", WRONG_RELEASE_FILENAMES)
+def test_wrong_release_filename(product, filename):
+    assert not is_release_filename(product, filename)
```
Comment (leplatrem) [inline data_tools/tests/test_utils.py]: Nice tests 👍 👍

**Comment by Natim** [comment]
> What is the status of this PR? Is it supposed to replace #30?

It is not, we should merge #30 first.

Because you didn't want me to keep working on things, I created this PR to be able to move forward while we are reviewing #30.

**Comment by Natim** [comment]
> Personally I think we ought to keep the listen_pulse script, even if we can't use it yet

I don't agree, the code is accessible on Github and the plan is to port it to Python 3 as soon as we merge #30.

**Comment by Natim** [comment]
> I still wonder what was wrong with ui and jobs,

jobs is to restrictive for what the buildhub python package is meant to be.
The names are things that we brainstormed with @magopian while being in Paris. We find them more explicit and self documented than `ui` and `jobs` but I totally don't care about naming so if you feel that it doesn't add any value, we can keep the old names.

**Pull request merged**